pennylane/templates/layers/basic_entangler.py

```diff
@@ -157,14 +122,91 @@ def circuit(weights):
         ``ValueError: Wrong number of parameters``.
     """
 
-    if rotation is None:
-        rotation = RX
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(self, weights, wires=None, rotation=None, do_queue=True):
+
+        self.rotation = rotation or qml.RX
+
+        super().__init__(weights, wires=wires, do_queue=do_queue)
+        self._preprocess()
+
+    def expand(self):
+
+        weights = self.data[0]
+
+        # first dimension of the weights tensor determines
+        # the number of layers
+        repeat = qml.math.shape(weights)[0]
+
+        with qml.tape.QuantumTape() as tape:
+
+            for layer in range(repeat):
+                for i in range(len(self.wires)):
+                    self.rotation(weights[layer][i], wires=self.wires[i : i + 1])
+
+                broadcast(unitary=qml.CNOT, pattern="ring", wires=self.wires)
```

Comment:
I thought from a comment above that `broadcast` is inefficient? So why is it ok to use here?

Reply:
Good catch, this should not be here!